src/metabase/api/user.clj

```diff
@@ -301,6 +315,11 @@
   (api/let-404 [user-before-update (fetch-user :id id, :is_active true)]
     ;; Google/LDAP non-admin users can't change their email to prevent account hijacking
     (api/check-403 (valid-email-update? user-before-update email))
+    ;; SSO users (JWT, SAML, LDAP, Google) can't change their first/last names
+    (when (or (contains? body :first_name)
+              (contains? body :last_name))
+      (api/checkp (valid-name-update? user-before-update first_name last_name)
+        "name" (tru "Editing names is not allowed for SSO users.")))
```

Comment:
This call to `api/checkp` has `"name"` as its `field-name`. In this case, `field-name` can be used to highlight the form field that caused the an error in the FE. But since `"name"` isnt the name of one of the fields, that won't happen. 

Here it is with "name" as the `field-name`:
<img width="317" alt="Screen Shot 2022-07-07 at 3 36 27 PM" src="https://user-images.githubusercontent.com/343288/177876409-040c1abf-d642-4883-b311-0b18eba704ca.png">

Then again with "first name":
<img width="545" alt="Screen Shot 2022-07-07 at 3 36 50 PM" src="https://user-images.githubusercontent.com/343288/177876472-3d434bc7-b8c6-4994-bd9e-987beb5fcbf0.png">

I think highlighting the invalid field on FE in red only happens when `field-name` matches up with the specified field.

So you may want to split this up into 2 checks, one for `"first_name"` and one for `"last_name"`. 

Maybe something like:


```clojure
(when (contains? body :first_name)
  (api/checkp (valid-name-update? user-before-update first_name)
    "first_name" (tru "Editing first name is not allowed for SSO users.")))
(when (contains? body :last_name)
  (api/checkp (valid-name-update? user-before-update last_name)
    "last_name" (tru "Editing last name is not allowed for SSO users.")))
```

Of course you'll need to update `valid-name-update?` a bit too.

Reply:
Hmm, this is a great point. I'll make the change. Though, It is also the case that this shouldn't actually occur. We have (or are supposed to have) made a change that prevents the UI from even showing those fields when a user is an SSO user. Still, a worthwhile improvement, thanks :)